packages/skygear-core/lib/record.js

```diff
@@ -102,46 +97,69 @@ export default class Record {
       throw new Error(
         'RecordType is not valid. Please start with alphanumeric string.');
     }
-    if (!attrs) {
-      attrs = _.assign({}, defaultAttrs);
+
+    let {
+      _id,
+      _recordID,
+      _recordType,
+      ...otherAttrs
+    } = attrs || {};
+
+    // fallback to parse `_id` only when `_recordID` is unavailable
+    if (!_recordID && _id) {
+      [_recordType, _recordID] = Record.parseDeprecatedID(_id);
     }
-    this._recordType = recordType;
-    // Favouring `id`, since `id` will always contains type information if
-    // exist.
-    let id = attrs.id || attrs._id;
-    if (id === null || id === undefined) {
-      id = uuid.v4();
-    } else {
-      let [type, name] = Record.parseID(id);
-      if (type !== this._recordType) {
-        throw new Error('_id is not valid. RecordType mismatch.');
-      }
-      id = name;
+
+    if (_recordType && _recordType !== recordType) {
+      throw new Error(
+        '_recordType in attributes does not match the constructor recordType'
+      );
     }
-    delete attrs.id; // because `id` is a readonly property
-    this._id = id;
+
+    this._recordType = recordType;
+    this._recordID = _recordID || uuid.v4();
+
     this._access = null;
-    this.update(attrs);
-    this.updateTransient(attrs._transient);
+    this.update(otherAttrs);
+    this.updateTransient(otherAttrs._transient);
   }
 
   /**
+   * Type of the record.
+   *
    * @type {String}
    */
   get recordType() {
     return this._recordType;
   }
 
   /**
-   * Record id in the format of `type/id`
+   * ID of the record.
    *
    * @type {String}
    */
+  get recordID() {
+    return this._recordID;
+  }
+
+  /**
+   * ID of the record in the deprecated format (i.e. `type/id`).
+   *
+   * @type {String}
+   *
+   * @deprecated Use `recordType` and `recordID` instead.
+   */
   get id() {
-    return this._recordType + '/' + this._id;
+    console.warn(
```

Comment:
I just came across this https://www.npmjs.com/package/util-deprecate. If we use it the warning will be shown only once.